BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -99,6 +99,76 @@ PrepareStage2 (
   return Dst;
 }
 
+/**
+  Load, verify and append a hashes from external hash store compoment.
+
+  @param[in] LdrGlobal   Pointer to loader global data.
+
+  @retval   EFI_SUCCESS             Hash store was appended successfully.
+  @retval   EFI_NOT_FOUND           Could not locate the external hash store.
+  @retval   EFI_OUT_OF_RESOURCES    Not enough buffer to load the external hash store.
+  @retval   EFI_UNSUPPORTED         Hash store format is not valid.
+  @retval   EFI_SECURITY_VIOLATION  Hash store authentication failed.
+**/
+EFI_STATUS
+AppendHashStore (
+  IN LOADER_GLOBAL_DATA       *LdrGlobal
+  )
+{
+  EFI_STATUS           Status;
+  HASH_STORE_TABLE    *LdrKeyHashBlob;
+  HASH_STORE_TABLE    *OemKeyHashBlob;
+  HASH_STORE_TABLE    *OemKeyHashComp;
+  UINT32               OemKeyHashCompBase;
+  UINT32               OemKeyHashUsedLength;
+  INT32                KeyHashSize;
+  UINT8                AuthInfo[RSA_SIGNATURE_AND_KEY_SIZE];
+  SIGNATURE_HDR       *SignHdr;
+  PUB_KEY_HDR         *PubKeyHdr;
+
+  Status = GetComponentInfo (FLASH_MAP_SIG_KEYHASH, &OemKeyHashCompBase, NULL);
+  if (EFI_ERROR(Status)) {
+    return EFI_NOT_FOUND;
+  }
+
+  // Check used length before copying to temporary memory
+  OemKeyHashComp = (HASH_STORE_TABLE *) OemKeyHashCompBase;
+  LdrKeyHashBlob = (HASH_STORE_TABLE *) LdrGlobal->HashStorePtr;
+  OemKeyHashUsedLength  = OemKeyHashComp->UsedLength;
+  if (OemKeyHashUsedLength > LdrKeyHashBlob->TotalLength - LdrKeyHashBlob->UsedLength) {

```

Comment:
Do we have room for OemKeyHash? Does the HashStore already have HASH_STORE_MAX_IDX_NUM 5 of hashes?

Reply:
yes, we will have space to add new hash.  the buffer size is not exactly 5 hash entry, it is determined by PcdHashStoreSize.

Reply:
Got it. Thanks.